src/main/controllers/features/sole-trader/correspodanceAddressDetailsController.ts

```diff
@@ -49,24 +50,9 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
                 addresses: acspData?.addresses
             });
         } else {
-            const addressList = acspData.addresses!;
-            const selectPremise = req.body.correspondenceAddress;
-            for (const ukAddress of addressList) {
-                if (ukAddress.propertyDetails!.toUpperCase() === selectPremise.toUpperCase()) {
-                    // Save the correspondence address to session
-                    const correspondenceAddress = {
-                        propertyDetails: ukAddress.propertyDetails,
-                        line1: ukAddress.line1,
-                        line2: ukAddress.line2,
-                        town: ukAddress.town,
-                        country: ukAddress.country,
-                        postcode: ukAddress.postcode
-                    };
-                    saveDataInSession(req, CORRESPONDENCE_ADDRESS, correspondenceAddress);
-
-                }
 
-            }
+            const selectPremise = req.body.correspondenceAddress;
+            CorrespondenceAddressDetailsService.saveCorrespondenceDetailsAddress(req, acspData, selectPremise);
```

Comment:
Do we not have to create an object of the service class first and then call the function?